### PR TITLE
Fix nightly test errors

### PR DESCRIPTION
**Pull request by loomlike**
### Description
Fix nightly test errors

### Related Issues
#2041 
#2035 

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [x] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: todo.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+https://huggingface.co/datasets/amazon_reviews_multi
```
Comment (miguelgfierro) [inline todo.txt:1]: I would recommend to remove this and add it to the issues

**Review by miguelgfierro — COMMENTED** [review]
Looks good, I have some questions

File: recommenders/utils/notebook_utils.py [hunk 2]
```diff
@@ -44,6 +44,28 @@ def is_databricks():
         return False
 
 
+def update_parameters(parameter_cell_source, new_parameters):
```
[the remark this answers is not in the document]
Comment (loomlike) [reply recommenders/utils/notebook_utils.py:47]: private sounds good. what's our convention? double prefix underscore or single? `__update_parameters` vs `_update_parameters`

File: tests/unit/examples/test_notebooks_pyspark.py [hunk 3]
```diff
@@ -96,7 +96,11 @@ def test_evaluation_diversity_runs(notebooks, output_notebook, kernel_name):
     )
 
 
-# This is a flaky test that can fail unexpectedly
+# mock100 dataset throws the following error:
+#   TrainValidationSplit IllegalArgumentException: requirement failed:
+#   Nothing has been added to this summarizer.
+# This seems to be caused by cold start problem -- https://stackoverflow.com/questions/58827795/requirement-failed-nothing-has-been-added-to-this-summarizer
+# In terms of the processing speed at Spark, "100k" dataset does not take much longer than "mock100" dataset and thus use "100k" here to go around the issue.
```
[the remark this answers is not in the document]
Comment (loomlike) [reply tests/unit/examples/test_notebooks_pyspark.py:103]: >I'm curious, why mock100 is not much faster than 100k. Are they the same size?
not same. mock100 is generating 100 samples, which takes more time in spark than downloading 100k data.
Also in spark. 100 vs 100k are about the same, both tiny. I tested sum("rating") and both took around 0.1 sec.

>if we modify mock100 will it be worth it in comparison to leave it with 100k?
We randomly generate records based on the schema. We may try stratified split to make sure all the users & items in test set are in train set, but again 100 vs 100k is about the same.

File: todo.txt (hunk 1 above)
Comment (loomlike) [reply todo.txt:1]: accidently added!

File: tests/unit/examples/test_notebooks_python.py [hunk 4]
```diff
@@ -83,6 +83,7 @@ def test_cornac_deep_dive_runs(notebooks, output_notebook, kernel_name):
 
 @pytest.mark.notebooks
 @pytest.mark.experimental
+@pytest.mark.skip(reason="rlrmc doesn't work with any officially released pymanopt package")
```
[the remark this answers is not in the document]
Comment (loomlike) [reply tests/unit/examples/test_notebooks_python.py:86]: fyi, `test_geoimc_functional` at the functional test also suffers the same issue.

**Comment by loomlike** [comment]
New commits  fix functional test errors too. Only two functional tests, benchmark_gpu and benchmark_cpu, are failing now:

NCF's ndcg has been changed somehow. Need to investivate.
```
test_benchmark_movielens_gpu  
...
E             Obtained: [0.41627350482234177]
E             Expected: 0.382793 ± 5.0e-02
```
BPR's ndcg has been changed too...
```
test_benchmark_movielens_cpu
...
E             Obtained: [0.44499071687886577]
E             Expected: 0.094444 ± 5.0e-02
```
```

File: recommenders/utils/notebook_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (miguelgfierro) [reply recommenders/utils/notebook_utils.py:47]: Just `_update_parameters`

File: tests/unit/examples/test_notebooks_pyspark.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (miguelgfierro) [reply tests/unit/examples/test_notebooks_pyspark.py:103]: got it, thanks

**Review by miguelgfierro — APPROVED** [review]

**Comment by loomlike** [comment]
> New commits fix functional test errors too. Only three functional tests, sasrec, benchmark_gpu and benchmark_cpu, are failing now:
> 
> SASREC's hit@10
> 
> ```
> E             Obtained: 0.3702
> E             Expected: 0.4244 ± 5.0e-02
> ```
> 
> NCF's ndcg (from test_benchmark_movielens_gpu)
> 
> ```
> E             Obtained: [0.41627350482234177]
> E             Expected: 0.382793 ± 5.0e-02
> ```
> 
> BPR's ndcg (from test_benchmark_movielens_cpu)
> 
> ```
> E             Obtained: [0.44499071687886577]
> E             Expected: 0.094444 ± 5.0e-02
> ```

fixed those issues.

**Comment by miguelgfierro** [comment]
this is great @loomlike, if the PR is ready, do you want to merge?

**Pull request merged**